### ISSUE TITLE
Add `userID`/`connectionID`

### ISSUE THREAD

**Issue by stress-tess** [Bears-R-Us/arkouda#1899]
Part of #1898:

Assign each connection a `userID` or `connectionID`. (It seems feasible to allow users to set their own `userID` after a uniqueness check against a list stored in ServerConfig, but I think we should hold off on that for now). This information will be added to all objects in the symbol table, so each object will know which user created itself. We will incorporate the `userID` into the registered name. Initially we allow anyone to attach to any registered object regardless of who created it, but they can only modify the objects that they created. (Eventually we will likely want to allow toggling of data sharing on a per object or per connection basis)

This will facilitate collaboration between users

  - User1 can share a pdarray containing interesting results with User2 by providing the registered name. User2 can then attach to that pdarray

But safeguard the data

  - User2 can't modify the underlying data within the pdarray without creating their own copy, which protects User1 from having the pdarray change from underneath them without their knowledge

We will need to add checks to existing registration functions to account for this new functionality (`unregister`, `list_registry`, ...)

Additionally the first connection to the server will be special (perhaps named admin or something similar) and objects registered by it will be easily visible by all users via a function call (something like `list_preloaded_data` that will behave like `list_registry` or `list_symbol_table`). This will allow an admin to pre-load objects that all team members can attach to at the beginning of the day and create their own copies without needing to load themselves.

**Comment by Ethan-DeBandi99** [comment]
I think this looks good. The only thing that concerns me is adding the userID to the name when registering since that probably won't be user facing. I would venture to say we could get the same functionality with just storing it in the SymEntry.

**Comment by pierce314159** [comment]
> I think this looks good. The only thing that concerns me is adding the userID to the name when registering since that probably won't be user facing. I would venture to say we could get the same functionality with just storing it in the SymEntry.

yeah that makes sense. I was thinking it would be easier to implement `list_preloaded_data` with the `userID` in the name, but I think it should be easy enough to just keep a list on the server for the names registered by admin. I was also planning on having `list_registry` only show the objects that a user and admin has registered to make things less clunky, I'm not sure if that's doable without it but im also not sure it's necessary especially right away
